### PR TITLE
Cherry pick GCC8 fixes for nvcc_wrapper

### PR DESCRIPTION
**Pull request by jjwilke**

### CODE REVIEW

**Comment by jjwilke** [comment]
See https://github.com/kokkos/kokkos/pull/2643

**Comment by ndellingwood** [comment]
@jjwilke @masterleinad  does this need to be merged ASAP for progress on #2637 and #2645 or should I hold off? I don't want to disrupt your workflow, it's unclear to me if these changes are necessary for the other PRs or orthogonal to the issues they are meant to address.

**Comment by jjwilke** [comment]
It is not necessary for the other PRs. These bugs showed up during the bootcamp and will need to be in the release.

File: bin/nvcc_wrapper [hunk 1]
```diff
@@ -100,7 +100,7 @@ depfile_target_arg=""
 remove_duplicate_link_files=0
 
 function warn_std_flag() {
-  echo "nvcc_wrapper - *warning* you have set multiple optimization flags (-std=c++1* or --std=c++1*), only the first is used because nvcc can only accept a single std setting"
+  echo "nvcc_wrapper - *warning* you have set multiple standard flags (-std=c++1* or --std=c++1*), only the last is used because nvcc can only accept a single std setting"
```
Comment (masterleinad) [inline bin/nvcc_wrapper]: This line is different from `develop`, see https://github.com/kokkos/kokkos/blob/9208fb959c6fc93d7ce727b7b6082a0cfabd5122/bin/nvcc_wrapper#L103.

**Pull request closed without merging**